lib/avo/fields/concerns/file_authorization.rb

```diff
@@ -0,0 +1,25 @@
+module Avo
+  module Fields
+    module Concerns
+      module FileAuthorization
+        extend ActiveSupport::Concern
+
+        delegate :authorization, to: :@resource
+        delegate :authorize_action, to: :authorization
+        delegate :id, :model, to: :@field
+
+        # Dynamically generate can_upload_file?, can_delete_file?, and can_download_file? methods
+        [:upload, :delete, :download].each do |action|
```

Comment:
I appreciate attempting to DRY the code a bit, but I think that having searchable code will help our contributors and us in the long run. For example, you can't find these methods if you do a global search for `can_download_file?` in the project.

I'm find with a little duplication if we get all that power instead. What do you think?

Reply:
Sure, and I agree, this isn't searchable, i just love how much we can DRY the code with ruby and I let myself be carried away by the moment =')) I'll make it searchable and more readable on the next commit =)

Reply:
Hmm I just thanked about this, whenever we use `define_method` we should add an comment with `# def method_name(s)` this way will be found on a global search. In this case:
```ruby
# def can_download_file?
# def can_upload_file?
# def can_delete_file?
```

Anyway i already committed and removed the dynamic `define_method` here

Reply:
Yeah, that could work. I don't see any downsides to that.